### PR TITLE
[MM-20456] Typescript Conversion of integrations/edit_outgoing_webhook

### PR DESCRIPTION
**Pull request by Ayanrocks**
#### Summary
In this PR I have converted the integrations/edit_outgoing_webhook module to typescript as stated in the issue #15484 on github. This PR fixes any loosely typed variables and has implemented strict types to the areas required.

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-20456

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -13465,9 +13461,15 @@
       }
     },
     "icu4c-data": {
+<<<<<<< HEAD
```
Comment (hanzei) [inline package-lock.json:13464]: It seems like your committed some unrelated changes to this file

Comment (Ayanrocks) [reply package-lock.json:13464]: Yes. can you tell me how to remove the package.json file from this PR.

Comment (Ayanrocks) [reply package-lock.json]: Never mind I have removed the file. Please check now. @hanzei

**Comment by Ayanrocks** [comment]
Any review updates? @hanzei @nevyangelova @devinbinnie

**Comment by nevyangelova** [comment]
Hey @Ayanrocks LGTM! Can you please update the snapshots?

File: components/integrations/edit_outgoing_webhook/edit_outgoing_webhook.tsx [hunk 2]
```diff
@@ -14,105 +16,110 @@ const HEADER = {id: 'integrations.edit', defaultMessage: 'Edit'};
...
+    confirmModalDismissed = (): void => {
         this.setState({showConfirmModal: false});
     }
 
-    submitHook = async () => {
+    submitHook = async (): Promise<void> => {
         this.setState({serverError: ''});
 
-        const {data, error} = await this.props.actions.updateOutgoingHook(this.newHook);
+        const {data, error}: any = await this.props.actions.updateOutgoingHook(this.newHook!);
```
Comment (devinbinnie) [inline components/integrations/edit_outgoing_webhook/edit_outgoing_webhook.tsx:122]: Is it possible to type this as `{data: OutgoingWebhook, error: ServerError}`?

File: components/integrations/edit_outgoing_webhook/index.ts [hunk 3]
```diff
@@ -2,29 +2,37 @@
...
 import {connect} from 'react-redux';
-import {bindActionCreators} from 'redux';
+import {bindActionCreators, Dispatch} from 'redux';
 import {getOutgoingHook, updateOutgoingHook} from 'mattermost-redux/actions/integrations';
 import {getConfig} from 'mattermost-redux/selectors/entities/general';
+import {GlobalState} from 'mattermost-redux/types/store';
+import {GenericAction} from 'mattermost-redux/types/actions';
 
-import EditOutgoingWebhook from './edit_outgoing_webhook.jsx';
+import EditOutgoingWebhook from './edit_outgoing_webhook.js';
```
Comment (devinbinnie) [inline components/integrations/edit_outgoing_webhook/index.ts:11]: You don't need to add the `.js` to the end of this.

**Review by devinbinnie — CHANGES_REQUESTED** [review]
Thanks for the contribution @Ayanrocks!
I've left a few comments for your review.

File: components/integrations/edit_outgoing_webhook/edit_outgoing_webhook.tsx (hunk 2 above)
Comment (Ayanrocks) [reply components/integrations/edit_outgoing_webhook/edit_outgoing_webhook.tsx:122]: This result in the following error

```
TS2739: Type 'ActionFunc' is missing the following properties from type '{ data: OutgoingWebhook; error: ServerError; }': data, error
```
the `ActionFunc` returns `ActionResult` and the type definition is this
```
export type ActionResult = {
    data: any;
} | {
    error: any;
};
```

Comment (devinbinnie) [reply components/integrations/edit_outgoing_webhook/edit_outgoing_webhook.tsx:123]: So what you can do here is use the `ActionCreatorsMapObject` in the redux connector to manually define the return type. Usually that will fix this.

**Comment by nevyangelova** [comment]
Thanks @Ayanrocks , you just have a lint error `/root/mattermost/mattermost-webapp/components/integrations/edit_outgoing_webhook/edit_outgoing_webhook.tsx
  9:9  error  'ServerError' is defined but never used  @typescript-eslint/no-unused-vars`

**Review by nevyangelova — APPROVED** [review]

File: components/integrations/edit_outgoing_webhook/edit_outgoing_webhook.tsx (hunk 2 above)
Comment (Ayanrocks) [reply components/integrations/edit_outgoing_webhook/edit_outgoing_webhook.tsx:123]: I'm sorry I didn't get that. Could you elaborate or connect with me in the mattermost developer community for a brief?

Comment (devinbinnie) [reply components/integrations/edit_outgoing_webhook/edit_outgoing_webhook.tsx:123]: Sent a message in MM :)

**Comment by Ayanrocks** [comment]
@devinbinnie

File: components/integrations/edit_outgoing_webhook/edit_outgoing_webhook.test.tsx [hunk 4]
```diff
@@ -2,17 +2,21 @@
...
+import {shallow, ShallowWrapper} from 'enzyme';
+
+import {OutgoingWebhook} from 'mattermost-redux/types/integrations';
+
+import EditOutgoingWebhook
+    from 'components/integrations/edit_outgoing_webhook/edit_outgoing_webhook';
 
 import {browserHistory} from 'utils/browser_history';
-import EditOutgoingWebhook from 'components/integrations/edit_outgoing_webhook/edit_outgoing_webhook.jsx';
+import {TestHelper} from '../../../utils/test_helper';
```
Comment (devinbinnie) [inline components/integrations/edit_outgoing_webhook/edit_outgoing_webhook.test.tsx:13]: Nit: Shouldn't need a relative path here, `'utils/test_helper'` should work just fine.

**Review by devinbinnie — APPROVED** [review]
LGTM! Just one non-blocking comment.
Thanks @Ayanrocks!

File: components/integrations/edit_outgoing_webhook/edit_outgoing_webhook.test.tsx (hunk 4 above)
Comment (Ayanrocks) [reply components/integrations/edit_outgoing_webhook/edit_outgoing_webhook.test.tsx]: fixed it

**Review by srkgupta — APPROVED** [review]
Did some regression testing on Outgoing Webhook page and there were no issues found. Approving the PR. Thanks @Ayanrocks
